initramfs/scripts/local-premount/resize

```diff
@@ -31,24 +31,24 @@ done
 
 writable_part="$(findfs LABEL=writable)"
 
-syspath="$(dirname $(realpath /sys/class/block/$(basename $writable_part)))"
-device="$(realpath /dev/block/$(cat $syspath/dev))"
-partition=$(cat $syspath/$(basename $writable_part)/partition)
+syspath="$(dirname "$(realpath /sys/class/block/"$(basename "$writable_part")")")"
+device="$(realpath /dev/block/"$(cat "$syspath"/dev)")"
+partition=$(cat "$syspath"/"$(basename "$writable_part")"/partition)
 
-device_size="$(($(cat $syspath/size)/2))"
-sum_size="$(($(grep $(basename $device)[a-z0-9] /proc/partitions|\
+device_size="$(($(cat "$syspath"/size)/2))"
+sum_size="$(($(grep "$(basename "$device")[a-z0-9]" /proc/partitions|\
     tr -s ' '|cut -d' ' -f4|tr '\n' '+'|sed 's/+$//')))"
 
 get_end()
 {
     NUM=$1
     lastpart="$(grep -cE ^'[0-9]{1,}': $TMPFILE)"
     if [ "$lastpart" = "$NUM" ]; then
-        endsize="$(parted -ms $DEV print| grep ^/ | cut -d: -f2)"
+        endsize="$(parted -ms "$DEV" print| grep ^/ | cut -d: -f2)"
     else
         # we are not at the end ! get the start of the next partition
         # (minus 1 byte) instead of using the absolute end of the disk
-        endsize=$(($(parted -ms $DEV unit B print|grep ^$(($num+1)):|\
```

Comment:
yes, $num never existed, and likewise we do not have any image yet where the "else" would actually be in effect (only if "writable" is not the last partition on disk, we do not have such images yet)